src/controllers/dynakube/dynakube_controller.go

```
@@ -174,183 +170,172 @@ func (controller *DynakubeController) reconcileIstio(dynakube *dynatracev1beta1.
-		).Reconcile(ctx, dkState)
-		if dkState.Error(err) {
-			return nil
+func (controller *DynakubeController) reconcileOneAgent(ctx context.Context, dynakube *dynatracev1beta1.DynaKube) (err error) {
+	if dynakube.HostMonitoringMode() {
+		err = oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, daemonset.DeploymentTypeHostMonitoring,
+		).Reconcile(ctx, dynakube)
+		if err != nil {
+			return err
 		}
-		dkState.Update(upd, "host monitoring reconciled")
-	} else if dkState.Instance.CloudNativeFullstackMode() {
-		upd, err := oneagent.NewOneAgentReconciler(
-			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.DeploymentTypeCloudNative,
-		).Reconcile(ctx, dkState)
-		if dkState.Error(err) {
-			return nil
+		log.Info("reconciled host-monitoring")
+	} else if dynakube.CloudNativeFullstackMode() {
+		err = oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, daemonset.DeploymentTypeCloudNative,
+		).Reconcile(ctx, dynakube)
+		if err != nil {
+			return err
 		}
-		dkState.Update(upd, "cloud native fullstack monitoring reconciled")
-	} else if dkState.Instance.ClassicFullStackMode() {
-		upd, err := oneagent.NewOneAgentReconciler(
-			controller.client, controller.apiReader, controller.scheme, dkState.Instance, daemonset.DeploymentTypeFullStack,
-		).Reconcile(ctx, dkState)
-		if dkState.Error(err) {
-			return nil
+		log.Info("reconciled cloud-native")
+	} else if dynakube.ClassicFullStackMode() {
+		err = oneagent.NewOneAgentReconciler(
+			controller.client, controller.apiReader, controller.scheme, daemonset.DeploymentTypeFullStack,
+		).Reconcile(ctx, dynakube)
+		if err != nil {
+			return err
 		}
-		dkState.Update(upd, "classic fullstack reconciled")
+		log.Info("reconciled classic-fullstack")
 	} else {
-		controller.removeOneAgentDaemonSet(dkState)
+		controller.removeOneAgentDaemonSet(ctx, dynakube)
```

do we have a log somewhere that we successfully removed the daemonset? also seems like we are ignoring any errors that may happen in this method